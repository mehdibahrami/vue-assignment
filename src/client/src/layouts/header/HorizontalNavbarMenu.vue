<template>
  <div class="navbar-horizontal-menu">
    <!-- Menu Items -->
    <div
      v-for="(item, index) in items"
      :key="index"
      class="navbar-horizontal-items"
    >
      <a :href="item.url" class="navbar-horizontal-selected-button">
        {{ item.title }}
      </a>
    </div>
  </div>
</template>

<script>
export default {
  name: "HorizontalNavbarMenu",
  props: {
    items: { type: Array, required: true },
  },
};
</script>

<style scoped lang="scss">
.navbar-horizontal-menu {
  display: none;
}

.navbar-horizontal-items {
  margin-left: 2.5rem;
  display: flex;
}

.navbar-horizontal-selected-button {
  background-color: $green-dark;
  color: $white;
  padding: 0.5rem 0.75rem 0.5rem 0.75rem;
  border-radius: 0.375rem;
  font-size: $font-size-sm;
  line-height: 1.25rem;
}

.navbar-horizontal-selected-button:hover {
  background-color: $green-darker;
}

@media (min-width: 768px) {
  .navbar-horizontal-menu {
    display: block;
  }
}
</style>
