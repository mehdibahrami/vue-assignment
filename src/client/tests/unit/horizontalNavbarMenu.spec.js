import { shallowMount } from "@vue/test-utils";
import HorizontalNavbarMenu from "@/layouts/header/HorizontalNavbarMenu.vue";

describe("HorizontalNavbarMenu", () => {
  let wrapper;

  it("renders correctly", () => {
    // shallow mounting the main component
    wrapper = shallowMount(HorizontalNavbarMenu, {
      props: {
        items: [
          {
            url: "/",
            title: "Dashboard",
          },
        ],
      },
    });

    expect(
      wrapper.find("a.navbar-horizontal-selected-button").attributes(["href"])
    ).toBe("/");
    expect(wrapper.find("a.navbar-horizontal-selected-button").text()).toBe(
      "Dashboard"
    );
    expect(wrapper.element).toMatchSnapshot();
  });
});
