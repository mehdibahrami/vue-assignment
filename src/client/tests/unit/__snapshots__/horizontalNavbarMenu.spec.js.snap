// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalNavbarMenu renders correctly 1`] = `
<div
  class="navbar-horizontal-menu"
>
  <!-- Menu Items -->
  
  <div
    class="navbar-horizontal-items"
  >
    <a
      class="navbar-horizontal-selected-button"
      href="/"
    >
      Dashboard
    </a>
  </div>
  
</div>
`;
